mod-di-converter-storage-server/src/main/resources/templates/db_scripts/associations-migration/init_wrappers.sql

```diff
@@ -0,0 +1,275 @@
+                                and mh.jsonb ->> 'jobProfileId' = m.jsonb ->> 'jobProfileId')
+
+          select *
+          from match_hierarchy
+          LOOP
+            -- create wrapper for detail match profile
+            insert into profile_wrappers (id, profile_type, match_profile_id, associated_job_profile_id)
+            values (public.uuid_generate_v4(), 'MATCH_PROFILE',
+                    (recursive_record.jsonb ->> 'detailProfileId')::uuid,
+                    (recursive_record.jsonb ->> 'jobProfileId')::uuid)
+            returning id into detail_match_wrapper_id;
+
+            -- get existing wrapper for master match profile
+            select id
+            into master_match_wrapper_id
+            from profile_wrappers
+            where match_profile_id = (recursive_record.jsonb ->> 'masterProfileId')::uuid
+              and associated_job_profile_id = (recursive_record.jsonb ->> 'jobProfileId')::uuid;
+
+            -- update wrapper references
+            UPDATE match_to_match_profiles
+            SET jsonb = jsonb_set(
+              jsonb_set(jsonb, '{masterWrapperId}', to_jsonb(master_match_wrapper_id), true),
+              '{detailWrapperId}', to_jsonb(detail_match_wrapper_id), true)
+            WHERE id = recursive_record.id;
+          end loop;
+
+      end loop;
+    raise notice 'PROFILES_MIGRATION:: updated match_to_match_profiles';
+  end
+$$;
+
+DO
+-- match_to_action_profiles: create wrappers for match profiles in match_to_action_profiles associations
+$$
+  DECLARE
+    r                 record;
+    match_wrapper_id  UUID;
+    action_wrapper_id UUID;
+  BEGIN
+    FOR r IN
+      select mta.id, mta.jsonb
+      from match_to_action_profiles mta
+      LOOP
+        -- get existing wrapper for action profile
+        select id
+        into action_wrapper_id
+        from profile_wrappers
+        where action_profile_id = (r.jsonb ->> 'detailProfileId')::uuid;
+
+        -- get existing wrapper for match profile
+        select id
+        into match_wrapper_id
+        from profile_wrappers
+        where match_profile_id = (r.jsonb ->> 'masterProfileId')::uuid
+          and associated_job_profile_id = (r.jsonb ->> 'jobProfileId')::uuid;
+
+        if match_wrapper_id is null or action_wrapper_id is null then
+          raise warning 'BAD DATA, match_to_action_profiles id: %', r.id;
+          continue;
+        end if;
+
+        -- update wrapper references
+        UPDATE match_to_action_profiles
+        SET jsonb = jsonb_set(jsonb_set(jsonb, '{masterWrapperId}', to_jsonb(match_wrapper_id), true),
+                              '{detailWrapperId}', to_jsonb(action_wrapper_id), true)
+        WHERE id = r.id;
+      END LOOP;
+    RAISE NOTICE 'PROFILES_MIGRATION:: updated match_to_action_profiles';
+  END
+$$;
+
+DO
+-- action_to_action_profiles: use existing wrappers for action profiles in action_to_actions_profiles associations
```

Comment:
possibly we can omit this query because having action-to-action associations is not expected in practice